### DISCUSSION TITLE
Strongly want to support variable names other than English

### DISCUSSION THREAD

**Discussion by FishConsole** [beeware/briefcase#1463] [Ideas]
I am trying to use the Beeware you developed to encapsulate my backend, so that it can run normally on Android phones. During the encapsulation process, I did not create it through new, create and other methods. Instead, I directly created a single file and started the development. So I just took this template and did a simple makeover and ended it, so when I wanted to package it I was forced to do a makeover of the entire back end to make it meet Beeware's structural requirements, but when I packaged it I found that it seemed to be very unfriendly to languages other than English, I don't know why, I added the code modified to gbk in the python code, and added "# -*- coding: gbk -*-" in the first line to meet the coding needs, but still did not solve the problem. And when I run my single file directly, it is very normal to run, this is the biggest problem for me is that I can not package, dev command can not be used, a package can not be found, the following is my code, a person blood request to adapt, I spent 3 days did not solve this matter

---

```python
print("Current run code：", sys.getdefaultencoding())
# 示例：将字符串从UTF-8编码转换为另一种编码（例如GBK）
# Example: Convert a string from UTF-8 encoding to another encoding (e.g. GBK)
original_string = "你好"
utf8_encoded = original_string.encode('utf-8')
gbk_encoded = utf8_encoded.decode('utf-8').encode('gbk')
print("Raw string：", original_string)
print("UTF-8 string：", utf8_encoded)
print("GBK string：", gbk_encoded.decode('gbk'))
```

---

#### This is running a single file directly from the console
```
Current run code： utf-8
Raw string： 你好
UTF-8 string： b'\xe4\xbd\xa0\xe5\xa5\xbd'
GBK string： 你好
```

#### This is shown when you run the project using the "briefcase dev" command
```
PS E:\github\helloworld> briefcase dev
[helloworld] Starting in dev mode...
===========================================================================
Current run code锛\x9a utf-8
Raw string锛\x9a 浣犲ソ
UTF-8 string锛\x9a b'\xe4\xbd\xa0\xe5\xa5\xbd'
GBK string锛\x9a 浣犲ソ
PS E:\github\helloworld> 
```
---
# This is the entire code and you can repeat the problem
```python
# -*- coding: gbk -*-
# from helloworld.app import main

############################################################

# 输出当前运行时编码
import sys

print("Current run code：", sys.getdefaultencoding())
# 示例：将字符串从UTF-8编码转换为另一种编码（例如GBK）
# Example: Convert a string from UTF-8 encoding to another encoding (e.g. GBK)
original_string = "你好"
utf8_encoded = original_string.encode('utf-8')
gbk_encoded = utf8_encoded.decode('utf-8').encode('gbk')
print("Raw string：", original_string)
print("UTF-8 string：", utf8_encoded)
print("GBK string：", gbk_encoded.decode('gbk'))


############################################################

# if __name__ == '__main__':
#     main().main_loop()

```
# 🥹💕

**Comment by mhsmith** [comment]
@rmartin16: This is on Windows, so could it be fixed by the work you did last week on #1407?

**Reply by rmartin16** [reply]
Yeah; the work in #1425 handles non-7bit-ASCII characters much better.

@FishConsole, the next release of Briefcase will include these fixes to print non-English characters to the console. You can also install Briefcase from the `main` branch if you want to experiment with the latest code now.

**Comment by FishConsole** [comment]
Y U I J  -  自动回复邮件已经收到了,但是现在我有些事,没法回你.你等我一会儿,我看到就回.
